### PR TITLE
PNDA-4430: Handle mapping of users to yarn queues

### PR DESCRIPTION
**Pull request by Hitesh-Maplelabs**
Problem Statement: 
Handle mapping of users to queues to give functional parity with Spark.

Prerequisites: 
PRs of below stories need to be merged before this PR.
PNDA-4427
PNDA-4428

Test details:
Verified changes on AWS setups:
UBUNTU - PICO -CDH & HDP
RHEL - PICO -CDH & HDP

### CODE REVIEW

File: salt/resource-manager/init.sls [hunk 1]
```diff
@@ -1,7 +1,11 @@
 {% set resource_manager_path = pillar['resource_manager']['path'] %}
 {% set map_file = resource_manager_path + "policies/map.txt" %}
 {% set policy_file_link = resource_manager_path + pillar['resource_manager']['policy_file'] %}
-{% set execs = [ 'spark-submit', 'spark-shell', 'pyspark', 'mapred', 'hive', 'beeline' ] %} 
+{% set execs = [ 'spark-submit', 'spark-shell', 'pyspark', 'mapred', 'hive', 'beeline', 'flink', 'pyflink.sh', 'start-scala-shell.sh', 'yarn-session.sh' ] %} 
+{% set flink_version = pillar['flink']['release_version'] %}
```
Comment (jgarnier) [inline salt/resource-manager/init.sls:5]: could you rebase your branch with develop as you did not have flink in your pillar ATM please

File: salt/resource-manager/init.sls [hunk 2]
```diff
@@ -40,7 +44,7 @@ resource-manager_log:
     - makedirs: True
     - dir_mode: 755
 
-resource-manager_spark_common_wrapper:
+resource-manager_spark_flink_common_wrapper:
```
Comment (jgarnier) [inline salt/resource-manager/init.sls:46]: why did you rename the state id with Flink, worth keeping the original one. thx

File: salt/resource-manager/init.sls [hunk 3]
```diff
@@ -1,7 +1,11 @@
 {% set resource_manager_path = pillar['resource_manager']['path'] %}
 {% set map_file = resource_manager_path + "policies/map.txt" %}
 {% set policy_file_link = resource_manager_path + pillar['resource_manager']['policy_file'] %}
-{% set execs = [ 'spark-submit', 'spark-shell', 'pyspark', 'mapred', 'hive', 'beeline' ] %} 
+{% set execs = [ 'spark-submit', 'spark-shell', 'pyspark', 'mapred', 'hive', 'beeline', 'flink', 'pyflink.sh', 'start-scala-shell.sh', 'yarn-session.sh' ] %} 
+{% set flink_version = pillar['flink']['release_version'] %}
+{% set pnda_home = pillar['pnda']['homedir'] %}
+{% set flink_real_dir = pnda_home + '/flink-' + flink_version %}
```
Comment (jgarnier) [inline salt/resource-manager/init.sls]: flink_real_dir is not used, could you just create directly flink_bin_dir then?

**Review by jgarnier — CHANGES_REQUESTED** [review]

File: salt/resource-manager/init.sls (hunk 2 above)
Comment (Hitesh-Maplelabs) [reply salt/resource-manager/init.sls:46]: i have renamed the state with flink, as it will be the wrapper for both  spark and flink. if you want so will rename it as original ?

File: salt/resource-manager/init.sls (hunk 1 above)
Comment (Hitesh-Maplelabs) [reply salt/resource-manager/init.sls:5]: Sure, will rebase the branch.

File: salt/resource-manager/init.sls (hunk 3 above)
Comment (Hitesh-Maplelabs) [reply salt/resource-manager/init.sls]: will remove the  flink_real_dir  variable.  thanks

File: salt/resource-manager/init.sls (hunk 2 above)
Comment (jgarnier) [reply salt/resource-manager/init.sls:46]: ok let's keep Flink too then

File: salt/resource-manager/init.sls (hunk 1 above)
Comment (Hitesh-Maplelabs) [reply salt/resource-manager/init.sls:5]: Done..

Comment (Hitesh-Maplelabs) [reply salt/resource-manager/init.sls:5]: Done
